### PR TITLE
Added documentation decorator

### PR DESCRIPTION
**Pull request by VictorCoder123**
It is a fully functional document decorator for model documentation, please consider to integrate it into WebGME.

### CODE REVIEW

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.css [hunk 1]
```diff
@@ -0,0 +1,45 @@
+/**
+ * @author rkereskenyi / https://github.com/rkereskenyi
```
Comment (lattmann) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.css]: Add your Github username

File: src/client/decorators/DocumentDecorator/Core/DocumentDecorator.Core.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+/**
+ * @author Qishen Zhang
```
Comment (lattmann) [inline src/client/decorators/DocumentDecorator/Core/DocumentDecorator.Core.js]: Please use your github username too.

File: src/client/decorators/DocumentDecorator/Core/DocumentDecorator.Core.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    	};
+
+    	/**
+    	 * Load EpicEditor after it is created, two steps need to be seperated, 
+    	 * because editor cannot be load before container element is not rendered
+    	 * on DOM element
+    	 * @return {[type]} [description]
+    	 */
+    	DocumentDecoratorCore.prototype.loadEditor = function(){
+    		this.editor.load()
```
Comment (lattmann) [inline src/client/decorators/DocumentDecorator/Core/DocumentDecorator.Core.js]: semicolon

**Comment by lattmann** [comment]
Please check in the scss files too, if you have them.

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
+/**
+ * @author Qishen Zhang
```
Comment (lattmann) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js]: Same as above. Please add your github username as a link. see other files.

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+		 * Initialize DocumentEditorDialog by creating EpicEditor in Bootstrap modal
+		 * window and set event listeners on its subcomponents like save button. Notice
+		 * EpicEditor is created but not loaded yet. The creation and loading of editor 
+		 * are seperated due to the reason decorator component is not appended to DOM within
+		 * its own domain.
+		 * @param  {String} 	text           Text to be rendered in editor initially
+		 * @param  {Function}   saveCallback   Callback function after click save button
+		 * @return {void}              
+		 */
+		DocumentEditorDialog.prototype.initialize = function(text, saveCallback){
```
Comment (lattmann) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js]: Space between `){`

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+			var self = this;
```
Comment (lattmann) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js]: Indentation is not right in this function.

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js (hunk 5 above)
Comment (VictorCoder123) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js]: I don't know why the indentation on Github is not consistent with sublime editor. All codes in my sublime editor is well indented with four space. I'll try to figure it out.

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+        }
+
+        return false;
+    };
+
+    /**
+     * Initialize Dialog and Editor creation
+     * @return {void} 
+     */
+    DocumentDecorator.prototype._initDialog = function(){
```
Comment (pmeijer) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js]: Missing spaces before and after the `function`

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+        var self = this;
+        var client = this._control._client;
+        var metaObj = client.getMeta(this._metaInfo[CONSTANTS.GME_ID]);
+        var nodeObj = client.getNode(this._metaInfo[CONSTANTS.GME_ID]);
+        var documentation = nodeObj.getAttribute('documentation') || 'Click to enter documentation.';
+        if(nodeObj.getAttribute('documentation') !== undefined){
```
Comment (pmeijer) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js]: missing spaces here too

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+            this.$doc.append($(marked(documentation)));
+        } 
+
+        // Initialize with documentation attribute and save callback function
+        this.editorDialog.initialize(documentation, 
+            function(text){
```
Comment (pmeijer) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js]: missing spaces

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (pmeijer) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentDecorator.DiagramDesignerWidget.js]: consider trimming these lines

File: src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+})
+
+
+
+
+
+
+
+
+
```
Comment (pmeijer) [inline src/client/decorators/DocumentDecorator/DiagramDesigner/DocumentEditorDialog.js]: trim

**Comment by pmeijer** [comment]
Qishen, your code looks good. Like the way it looked in action too. I have some minor style comments. If you have jssc activated in your editor they will show up as errors/warnings. If you can fix those before we pull, the git history won't be cluttered with other users.

**Comment by VictorCoder123** [comment]
Hi Patrik,
I think what you mean by "jssc" is to use .jshintrc file to lint code, and I have fixed all the style problems you mentioned. Take a look and please let me know if there is anything needs to be improved before pull.

**Pull request merged**